app/routes.py

```diff
@@ -1 +1,268 @@
+
+# - - - - - - - - - - - - - - - - - #
+###~~~--- goal model routes ---~~~###
+# - - - - - - - - - - - - - - - - - #
+
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@goal_bp.route("", methods=['POST'])
+def create_goal():
+    request_body = request.get_json()
+
+    try:
+        new_goal = Goal(
+            title = request_body["title"])
+    except:
+        abort(make_response({"details": f"Invalid data"}, 400))
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {"goal": new_goal.to_dict()}, 201
+
+@goal_bp.route("/<goal_id>/tasks", methods=['POST'])
+def assign_tasks_to_goal(goal_id):
+    request_body = request.get_json()
+    response_tasklist = request_body["task_ids"]
+
+    goal = validate_model(Goal, goal_id)
+
+    for task_id in response_tasklist:
+        single_task = validate_model(Task, task_id)
+        
+        if single_task:
+            goal.tasks.append(single_task)
+
+    db.session.commit()
+
+    return {
+        "id": goal.goal_id,
+        "task_ids": response_tasklist
+        }, 200
+
+@goal_bp.route("/<goal_id>/tasks", methods=['GET'])
+def read_goal_tasks(goal_id):
+    goal = validate_model(Goal, goal_id)
+    tasklist = []
+
+    for task in goal.tasks:
+        tasklist.append({
+            "id": task.task_id,
+            "goal_id": goal.goal_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": True if task.completed_at else False
+        })
+
+    return {
+        "id": goal.goal_id,
+        "title": goal.title,
+        "tasks": tasklist
+        }, 200
+
+@goal_bp.route("", methods=["GET"])
+def read_all_goals():
+    # querries
+    title_query = request.args.get("title")
+    sort_query = request.args.get("sort")
+
+    if title_query:
+        goals = Goal.query.filter_by(title=title_query)
+    
+    elif sort_query:
+        if sort_query == 'asc':
+            goals = Goal.query.order_by(Goal.title).all()
+        elif sort_query == 'desc':
+            goals = Goal.query.order_by(Goal.title.desc()).all()
+    
+    else:
+        goals = Goal.query.all()
+
+    goals_response = []
+
+    for goal in goals:
+        goals_response.append(goal.to_dict())
+
+    return jsonify(goals_response), 200
+
+@goal_bp.route("/<goal_id>", methods=["GET"])
+def read_one_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    return {"goal": goal.to_dict()}, 200
+
+@goal_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    request_body = request.get_json()
+
+    goal.title = request_body["title"]
+
+    db.session.commit()
+
+    return {"goal": goal.to_dict()}, 200
+
+@goal_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return jsonify({"details":f'Goal {goal.goal_id} \"{goal.title}\" successfully deleted'}), 200
```

Comment:
Well done on this project, I didn't have much to comment on and that is a good thing! Keep up the good work! Really looking forward to what you create in the frontend! Please feel free to reach out if you have any questions about the feedback that I left! ✨💫🤭 Great work, Winslow!